### PR TITLE
[SIW-1490] Add new function app support-func

### PR DESCRIPTION
**Pull request by silvicir**
#### Motivation and Context

A new function app in the wallet domain is needed for support purposes

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by grausof — COMMENTED** [review]
I am not aware of the context. I also ask @Krusty93 to do a review

**Comment by silvicir** [comment]
@mamu0 @christian-calabrese

File: infra/resources/_modules/function_apps/function_app_user.tf [hunk 1]
```diff
@@ -14,16 +14,16 @@ module "function_app_user" {
   health_check_path   = "/api/v1/wallet/health"
   node_version        = 20
 
-  subnet_cidr                          = var.cidr_subnet_user_func
+  subnet_cidr                          = var.cidr_subnet_func
```
Comment (Krusty93) [inline infra/resources/_modules/function_apps/function_app_user.tf:17]: The CIDR block should be different for each function as the module creates a new subnet using the block passed

**Comment by Krusty93** [comment]
Does one of the two function need to connect to external parties?

**Comment by silvicir** [comment]
> Does one of the two function need to connect to external parties?

no

File: infra/resources/_modules/function_apps/function_app_user.tf (hunk 1 above)
Comment (silvicir) [reply infra/resources/_modules/function_apps/function_app_user.tf:17]: ok. which cidr should I use?

Comment (Krusty93) [reply infra/resources/_modules/function_apps/function_app_user.tf:17]: `10.20.13.0`

**Comment by Krusty93** [comment]
> > Does one of the two function need to connect to external parties?
> 
> no

I still suggest to add the NAT Gateway integration with the new subnet:

```hcl
data "azurerm_nat_gateway" "nat" {
  name                            = "io-p-itn-ng-01"
  resource_group_name = "io-p-itn-common-rg-01"
}

resource "azurerm_subnet_nat_gateway_association" "func_support" {
  nat_gateway_id = azurerm_nat_gateway.nat.id
  subnet_id          = module.function_app_support.subnet.id
}
```

Please format the code and use string concatenation instead of using hardcoded values, but the code is actually this one

PR is ok though

File: infra/resources/prod/main.tf [hunk 2]
```diff
@@ -70,6 +70,7 @@ module "function_apps" {
   resource_group_name = azurerm_resource_group.wallet.name
 
   cidr_subnet_user_func                = "10.20.0.0/24"
+  cidr_subnet_support_func             = "10.20.13.0"
```
Comment (Krusty93) [inline infra/resources/prod/main.tf:73]:
```suggestion
  cidr_subnet_support_func             = "10.20.13.0/24"
```

**Review by Krusty93 — APPROVED** [review]

**Review by grausof — APPROVED** [review]

**Pull request merged**